docs/spec/spec.md

```diff
@@ -159,6 +159,11 @@ spec:
     metadata: ...
       labels:
         knative.dev/type: "function"  # One of "function" or "app"
+      annotations:
+        # +optional. When not specified, the revision can scale down to 0 pods
+        autoscaling.knative.dev/minReplicaCount: "2"  
```

Comment:
@cooperneil, what do you think about these annotations for providing Knative autoscaling bounds?

Reply:
Thanks @imikushin for doing this!  I think my suggestion to use the word "replica" is flawed.  Knative doesn't talk about "replicas", "instances", "clones" etc..  It only talks about "containers".  This was the reason I eventually chose the field name "ContainerConcurrency" in the Revision spec.

I would like to change these annotations to:
```
autoscaling.knative.dev/minContainerCount
autoscaling.knative.dev/maxContainerCount
```
What do you think?

Reply:
I didn't like the choice of names in the issue either, but thought I'd try to implement and see where it leads. I'd like to suggest: 
```
autoscaling.knative.dev/minScale
autoscaling.knative.dev/maxScale
```

If you insist on `(min|max)ContainerCount` I don't mind either. Let me know what you like more and I'll update the PR accordingly.

Reply:
Ping @josephburnett !

Reply:
🐸 @josephburnett @mattmoor 

I'd like to update this with whatever you guys decide

Reply:
I've chatted offline with @mattmoor about how to use these annotations, and I'll restate my understanding here.

The purpose of annotations (over the strongly-typed spec) is to allow users to reach under the covers and tune a component they know is there.  In this case, we want to tune the specific implementation of the KPA interface that comes with Knative.  The `autoscaling.knative.dev/class` identifies the particular KPA implementation being used and will remain with the KPA.  However other implementation-specific annotations such as `autoscaling.knative.dev/minScale` and `autoscaling.knative.dev/maxScale` will remain with the implementation, even if it is separated (at some later time.)

As such, my recommendation is to go ahead with the `autoscaling.knative.dev/minScale` and `autoscaling.knative.dev/maxScale` annotation names.  But do not document them in the Revision spec.  Document them somewhere else (maybe in the https://github.com/knative/serving/blob/master/pkg/autoscaler/README.md file).

Reply:
Done!